modules/transcription-service-ibm-watson-impl/src/main/java/org/opencastproject/transcription/ibmwatson/IBMWatsonTranscriptionService.java

```diff
@@ -872,16 +875,32 @@ public void run() {
 
       try {
         // Find jobs that are in progress and jobs that had transcription complete i.e. got the callback
-        List<TranscriptionJobControl> jobs = database.findByStatus(TranscriptionJobControl.Status.Progress.name(),
+
+        long providerId;
+        TranscriptionProviderControl providerInfo = database.findIdByProvider(PROVIDER);
+        if (providerInfo != null) {
+          providerId = providerInfo.getId();
+        } else {
+          logger.debug("No provider entry for {}", PROVIDER);
```

Comment:
I feel like this should be louder than debug.  If the dispatcher in the watson service can't find its own registration, something is wrong.

Reply:
Updated to warn. It should never happen in practice.